optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/forager/finalist/AbstractFinalistPodium.java

```diff
@@ -32,6 +32,8 @@
  */
 public abstract class AbstractFinalistPodium extends LocalSearchPhaseLifecycleListenerAdapter implements FinalistPodium {
 
+    protected static final int FINALIST_LIST_MAX_SIZE = 1_024_000;
```

Comment:
Any particular reason why so many finalists would be a good idea?

Reply:
No, but with a normal amount of memory we can handle this.
Figuring out the MAX_SIZE on which these so many finalist would be 100% of the time a bad idea, is impossible.
It would be a good guesstimate at best, so if we get it wrong: an potential risk for silent, surprising misbehaviour in how the algorithm reacts in certain conditions. 
This MAX_SIZE isn't meant to distinguish between good and bad. It's to distinguish between bad without OOME trouble and bad with OOME.